### PR TITLE
Ask for device when using adb with run-android

### PR DESCRIPTION
**Pull request by jreziga**
Sometimes adb see many devices (see: http://puu.sh/qoASz/c13f010dd0.jpg) and it can't `adb reverse` because of that (see: http://puu.sh/qoAR1/8412dfe25a.jpg)

So I've added the possibility to choose a device when you `run-android` your app. 

Final result: http://puu.sh/qoB12/329d5c27db.jpg

### CODE REVIEW

**Comment by facebook-github-bot-2** [comment]
By analyzing the blame information on this pull request, we identified @grabbou and @burgalon to be potential reviewers.

**Comment by facebook-github-bot-2** [comment]
Thank you for your pull request and welcome to our community. We require contributors to sign our Contributor License Agreement, and we don't seem to have you on file. In order for us to review and merge your code, please sign up at https://code.facebook.com/cla - and if you have received this in error or have any questions, please drop us a line at cla@fb.com. Thanks!

**Comment by facebook-github-bot-1** [comment]
Thank you for signing our Contributor License Agreement. We can now accept your code for this (and any) Facebook open source project. Thanks!

**Comment by facebook-github-bot-3** [comment]
@jreziga updated the pull request.

**Comment by facebook-github-bot-3** [comment]
@jreziga updated the pull request.

File: local-cli/runAndroid/runAndroid.js [hunk 1]
```diff
@@ -51,11 +47,16 @@ function getAdbPath() {
...
 
 // Runs ADB reverse tcp:8081 tcp:8081 to allow loading the jsbundle from the packager
-function tryRunAdbReverse() {
+function tryRunAdbReverse(device) {
   try {
     const adbPath = getAdbPath();
     const adbArgs = ['reverse', 'tcp:8081', 'tcp:8081'];
 
+    // If a device is specified then tell adb to use it
+    if (device !== undefined) {
```
Comment (ide) [inline local-cli/runAndroid/runAndroid.js]: Just `if (device)` is OK here

File: local-cli/runAndroid/runAndroid.js [hunk 2]
```diff
@@ -67,6 +68,22 @@ function tryRunAdbReverse() {
     console.log(chalk.yellow(
       `Could not run adb reverse: ${e.message}`
     ));
+
+    // Try again but specifying a device this time
+    if (device === undefined) {
```
Comment (ide) [inline local-cli/runAndroid/runAndroid.js]: `if (!device)`

File: local-cli/runAndroid/runAndroid.js [hunk 3]
```diff
@@ -67,6 +68,22 @@ function tryRunAdbReverse() {
...
+      const devices = adb.getDevices();
+      let choosenDevice = -1;
```
Comment (ide) [inline local-cli/runAndroid/runAndroid.js]: spelling: `chosenDevice`

File: local-cli/runAndroid/runAndroid.js [hunk 4]
```diff
@@ -67,6 +68,22 @@ function tryRunAdbReverse() {
...
+
+      console.log(chalk.green(
+          `\nChoose a device: \n${devices.map(function (device, index) { return `\t${index+1}. ${device}` }).join('\n')}\n`
```
Comment (ide) [inline local-cli/runAndroid/runAndroid.js]: Style: Use a short arrow function in map:
```js
devices.map((device, index) => `\t${index + 1}. ${device}`)
```

File: local-cli/runAndroid/runAndroid.js [hunk 5]
```diff
@@ -67,6 +68,22 @@ function tryRunAdbReverse() {
...
+      ));
+
+      while(isNaN(choosenDevice) || choosenDevice < 1 || choosenDevice > devices.length) {
```
Comment (ide) [inline local-cli/runAndroid/runAndroid.js]: Style: space between `while (`

File: local-cli/runAndroid/runAndroid.js [hunk 6]
```diff
@@ -67,6 +68,22 @@ function tryRunAdbReverse() {
...
+        choosenDevice = readlineSync.question('Your choice : ')
+      }
+
+      tryRunAdbReverse(devices[choosenDevice-1]);
```
Comment (ide) [inline local-cli/runAndroid/runAndroid.js]: Style: spaces around minus operator

Comment (jreziga) [inline local-cli/runAndroid/runAndroid.js]: Thanks for this code review !

**Comment by facebook-github-bot-8** [comment]
@jreziga updated the pull request.

**Comment by satya164** [comment]
Is it possible to run `adb-reverse` for each of the devices?

**Comment by jreziga** [comment]
Not a choice but a loop for all devices ?

**Comment by ekryski** [comment]
@satya164 I think you can and that would be helpful for sure but even this is a pretty nice improvement. imho pending those style issues being fixed this should totally 🚢

**Comment by facebook-github-bot-5** [comment]
@jreziga updated the pull request.

**Comment by facebook-github-bot-7** [comment]
@jreziga updated the pull request.

**Comment by facebook-github-bot-5** [comment]
@jreziga updated the pull request.

**Comment by facebook-github-bot-4** [comment]
@jreziga updated the pull request.

**Comment by satya164** [comment]
@jreziga Yeah, I'd prefer looping over the devices and running adb reverse on each of them.

@ekryski It is, but looping isn't a very big change, so I think we can update the PR to do that instead.

**Comment by facebook-github-bot-5** [comment]
@jreziga updated the pull request.

**Comment by facebook-github-bot-5** [comment]
@jreziga updated the pull request.

**Comment by jreziga** [comment]
So I edited my code for looping through all devices instead of asking for one device

**Comment by facebook-github-bot-5** [comment]
@jreziga updated the pull request.

File: local-cli/runAndroid/runAndroid.js [hunk 7]
```diff
@@ -15,11 +15,7 @@ const path = require('path');
 const isPackagerRunning = require('../util/isPackagerRunning');
 const Promise = require('promise');
 const adb = require('./adb');
-
-// Verifies this is an Android project
-function checkAndroid(root) {
-  return fs.existsSync(path.join(root, 'android/gradlew'));
-}
+const readlineSync = require('readline-sync');
```
Comment (satya164) [inline local-cli/runAndroid/runAndroid.js]: This doesn't seem to be used anymore

Comment (jreziga) [inline local-cli/runAndroid/runAndroid.js]: Yes you're right

**Comment by facebook-github-bot-5** [comment]
@jreziga updated the pull request.

**Comment by facebook-github-bot-5** [comment]
@jreziga updated the pull request.

**Comment by facebook-github-bot-6** [comment]
@jreziga updated the pull request.

**Comment by facebook-github-bot-2** [comment]
@jreziga updated the pull request.

**Comment by satya164** [comment]
@facebook-github-bot shipit

**Comment by aakashsigdel** [comment]
This commit removes the definition of `checkAndroid()` function on line 20, but it is still being called inside `runAndroid()`. Is this the wanted behavior? Because now calling react-native run-android, I get `checkAndroid is not defined`

**Comment by jreziga** [comment]
No idea when I've cloned the repository there wasn't that function. I noticed the problem but I suggested myself that there was an update.

I made an update, may I create a new pull request ?

**Comment by aakashsigdel** [comment]
I am not very sure about this. @satya164 will be more appropriate to comment on this.

**Pull request closed without merging**